ecl/hql/hqlfold.cpp

```diff
@@ -2154,6 +2154,74 @@ IHqlExpression * foldConstantOperator(IHqlExpression * expr, unsigned foldOption
             }
             break;
         }
+    case no_regex_findset:
+        {
+            IValue * t0 = expr->queryChild(0)->queryValue();
+            IValue * t1 = expr->queryChild(1)->queryValue();
+            if (t0 && t1)
+            {
+                if(isUnicodeType(t0->queryType()))
+                {
+                    unsigned plen = t0->queryType()->getStringLen();
+                    OwnedPtr<UChar> pattern (new UChar[plen+1]);
```

Comment:
This is cleaner.  I'm not sure how picky we should be.  In some ways I don't want to comment.  However technically something allocated with new X[n] should be deleted with delete [].  That class uses the non-array delete, so it could cause issues.  (Because UChar is a simple type I very much doubt it will.)
@richardkchapman what do you think about pickiness?
You could use OwnedMalloc<UChar> pattern(plen+1);
which is a bit strangely named, but would probably be better.